File: web/status-history/src/main/java/net/twisterrob/blt/gapp/view/handlebars/CustomHandlebarsViewsRenderer.java
package net.twisterrob.blt.gapp.view.handlebars;

import com.github.jknack.handlebars.Handlebars;

import io.micronaut.context.annotation.Replaces;
import io.micronaut.core.io.scan.ClassPathResourceLoader;
import io.micronaut.views.ViewsConfiguration;
import io.micronaut.views.handlebars.HandlebarsViewsRenderer;
import io.micronaut.views.handlebars.HandlebarsViewsRendererConfiguration;
import jakarta.inject.Singleton;

@Singleton
@Replaces(HandlebarsViewsRenderer.class)
public final class CustomHandlebarsViewsRenderer<T, R> extends HandlebarsViewsRenderer<T, R> {

	public CustomHandlebarsViewsRenderer(
			ViewsConfiguration viewsConfiguration,
			ClassPathResourceLoader resourceLoader,
			HandlebarsViewsRendererConfiguration handlebarsViewsRendererConfiguration,
			Handlebars handlebars
	) {
		super(viewsConfiguration, resourceLoader, handlebarsViewsRendererConfiguration, handlebars);

		this.handlebars.registerHelpers(HandlebarsHelpers.class);
	}
}

File: web/status-history/src/main/java/net/twisterrob/travel/statushistory/view/handlebars/CustomHandlebarsViewsRenderer.kt
package net.twisterrob.travel.statushistory.view.handlebars

import com.github.jknack.handlebars.Handlebars
import io.micronaut.context.annotation.Replaces
import io.micronaut.core.io.scan.ClassPathResourceLoader
import io.micronaut.views.ViewsConfiguration
import io.micronaut.views.handlebars.HandlebarsViewsRenderer
import io.micronaut.views.handlebars.HandlebarsViewsRendererConfiguration
import jakarta.inject.Singleton

@Singleton
@Replaces(HandlebarsViewsRenderer::class)
class CustomHandlebarsViewsRenderer<T, R>(
	viewsConfiguration: ViewsConfiguration,
	resourceLoader: ClassPathResourceLoader,
	handlebarsViewsRendererConfiguration: HandlebarsViewsRendererConfiguration,
	handlebars: Handlebars,
) : HandlebarsViewsRenderer<T, R>(
	viewsConfiguration,
	resourceLoader,
	handlebarsViewsRendererConfiguration,
	handlebars
) {

	init {
		this.handlebars.registerHelpers(HandlebarsHelpers::class.java)
	}
}
